blog_post_collaboration/spark_streaming_dataset.md

```diff
@@ -0,0 +1,322 @@
+
+```
+[03/31/23 11:23:46] INFO Loading data from 'raw.new_inventory' data_catalog.py:343
+ (SparkStreamingDataSet)... 
+
+[Stage 0:> (0 + 6) / 6]
+[03/31/23 11:23:49] INFO Running node: add_processing_time([raw.new_inventory]) node.py:329
+ -> [int.new_inventory] 
+ INFO Saving data to 'int.new_inventory' data_catalog.py:382
+ (SparkStreamingDataSet)... 
+23/03/31 11:23:49 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
+ INFO Completed 1 out of 5 tasks sequential_runner.py:85
+```
+
+The following log will be shown once Kedro has run through all the nodes and the full Spark execution plan has been created. This doesn't mean the stream processing has stopped as the post run hook keeps the Spark Session alive. 
+
+```
+INFO Pipeline execution completed successfully. runner.py:93
+```
+
+As new data comes in, new Spark logs will be shown, even after the "Pipeline execution completed" log.
+
+```
+[Stage 22:> (0 + 1) / 1]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (0 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (1 + 6) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (7 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (11 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:=> (14 + 7) / 200]
+```
+
+If there is an error in the input data, the Spark error logs will come through and Kedro will shut down the SparkContext and all the streams within it.
+
+```
+Caused by: java.lang.IllegalStateException: Cannot call methods on a stopped SparkContext.
+This stopped SparkContext was created at:
+org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
+sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
+sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
+sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
+java.lang.reflect.Constructor.newInstance(Constructor.java:423)
+py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
+py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
+py4j.Gateway.invoke(Gateway.java:238)
+py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
+py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
+py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
+py4j.ClientServerConnection.run(ClientServerConnection.java:106)
+java.lang.Thread.run(Thread.java:750)
+
+```
+
+
+## In summary
+
+In this article, we explained how to use a new Kedro dataset to create streaming pipelines. We created a new Kedro project using the Kedro `pyspark` starter and added two separate Hooks to the Kedro project to enable it to function as a streaming application, then configured the Kedro data catalog to use the new dataset, defining the streaming sources and sinks.
```

Comment:
Very basic summary as I moved what we had previously up to the top on @harisqb 's suggestion. Is it OK (particularly last section about Kafka)?

Reply:
Kafka part looks fine for me, as we have mentioned why it is not allowed currently :)